presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1338,7 +1338,7 @@ else if (u21 >= 0) {
             qhat = u21 / (v1 & LONG_MASK);
         }
         else {
-            qhat = divideUnsignedLong(u21, v1);
+            qhat = divideUnsignedLong(u21, v1 & LONG_MASK);
```

Comment:
is this change required by Hacker's Delight 9-3?

Reply:
The algorithm explained in the Hacker's Delight takes unsigned long as a divisor. For this very case it should always be `int`. So technically simple int can also be accepted. But it makes it a little bit confusing. Thus i decided to copy the algorithm verbatim.